sql/core/src/test/scala/org/apache/spark/sql/ApproximatePercentileQuerySuite.scala

```diff
@@ -182,7 +182,7 @@ class ApproximatePercentileQuerySuite extends QueryTest with SharedSparkSession
         spark.sql(
           s"""SELECT
              |  key,
-             |  percentile_approx(null, 0.5)
+             |  percentile_approx(cast(null as int), 0.5)
```

Comment:
Does it mean you can't write percentile_approx(null, 0.5) anymore? should that not be accepted?

Reply:
I have checked hive, `percentile_approx(null, 0.5)` is not valid. If it worth to keep back-compatibility of this, I can make NullType acceptable for percentile_approx

Reply:
It doesn't make much sense, yeah. I am not sure whether we generally return null or an error in a case like this. Following Hive seems like a reasonable guide. It could be OK to change behavior if there's a release note attached to the JIRA, and if we think it's more bug fix than anything

Reply:
Agreed. I'll updated this change in the PR description. And I guess the author's original purpose of these tests here are to verify the null handling of the non-null types, not `NullType`